### DISCUSSION TITLE
Understanding NextJS tutorial

### DISCUSSION THREAD

**Discussion by Jpac14** [vercel/next.js#18117] [Help]
Hi everyone,

I am very new to NextJS and I decided to start with the beginners tutorial on their [website](https://nextjs.org/learn/basics/create-nextjs-app?utm_source=next-site&utm_medium=nav-cta&utm_campaign=next-website).

I am up to lesson 3, section 8, where I mostly copy and paste the content, but I do understand it except for these few lines:
```
export default function Home() {
  return (
    <Layout home>
```

I understand that Layout takes a parameter home of any type but I don't understand where that parameter comes from, when I try to console.log it it says its undefined. Can somebody please explain what home is. I also know it used as a bool because they use a short if statement ( a ? b : c)

Thankyou in advance

**Comment by alanqchen** [comment]
Where are you doing the `console.log`? If you print the value of `home` in the `Layout` component itself, it should be a boolean (when true). In jsx both of the following are equivalent ways to pass a boolean prop for the compiler:
 
```js
<Layout home>
<Layout home={true}>
```
Similarly for false:
```js
<Layout> {/* home will be undefined */}
<Layout home={false}>
```

**Comment by AntCasson** [comment]
Hi OP,

After some googling, I found your question and thought I would explain it even tho the post is a bit dated. 

The `home` prop is added in the `index.js` file: 
```
export default function Home({ allPostsData }) {
  return (
    <Layout home>
... rest of code`

```

but in the `[id].js` file (later in the lessons) it is not: 
```
export default function Post({ postData }) {
  return (
    <Layout>
... rest of code

```

With the help of the `home` prop you can change the Layout component because it checks with a [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator).

When added (so the `home` prop returns true):
- the profile image is bigger
- the link `back to home` is not visible

When the `home` prop is left out (and returns false): 

- profile image is smaller
- `back to home` link is added at the bottom of the page.
